### PR TITLE
Added Javadoc to ClientConnectionManager

### PR DESCRIPTION
**Pull request by pveentjer**

### CODE REVIEW

File: hazelcast-client/src/main/java/com/hazelcast/client/connection/ClientConnectionManager.java [hunk 1]
```diff
@@ -26,25 +26,32 @@
...
      * @return connection if available, null otherwise
      */
     Connection getActiveConnection(Address address);
 
     /**
+     * Gets the Connection to the given member address if there already is a connection, or
+     * otherwise triggers the connection asynchronously and then null is returned.
+     *
      * @param address to be connected
      * @return associated connection if available, creates new connection otherwise
```
Comment (ihsandemir) [inline hazelcast-client/src/main/java/com/hazelcast/client/connection/ClientConnectionManager.java]: creates new connection and returns null

File: hazelcast-client/src/main/java/com/hazelcast/client/connection/ClientConnectionManager.java [hunk 2]
```diff
@@ -26,25 +26,32 @@
...
      *
      * @return true if alive, false otherwise.
      */
     boolean isAlive();
 
     /**
+     * Gets the Connection to the given member address.
+     *
+     * If address is null, null is returned.
+     *
```
Comment (ihsandemir) [inline hazelcast-client/src/main/java/com/hazelcast/client/connection/ClientConnectionManager.java]: thread-safe

File: hazelcast-client/src/main/java/com/hazelcast/client/connection/ClientConnectionManager.java [hunk 3]
```diff
@@ -58,11 +65,42 @@
      */
     Connection getOrTriggerConnect(Address address, boolean acquiresResource) throws IOException;
 
+    /**
+     * Gets all connections.
+     *
```
Comment (ihsandemir) [inline hazelcast-client/src/main/java/com/hazelcast/client/connection/ClientConnectionManager.java]: thread-safe

File: hazelcast-client/src/main/java/com/hazelcast/client/connection/ClientConnectionManager.java [hunk 4]
```diff
@@ -26,25 +26,32 @@
...
+ * Responsible for managing {@link com.hazelcast.client.connection.nio.ClientConnection} instances.
  */
 public interface ClientConnectionManager extends ConnectionListenable {
 
     /**
-     * Check if client connection manager is alive.
+     * Check this ClientConnectionManager is alive.
      * ClientConnectionManager is not alive only when client is closing.
      *
      * @return true if alive, false otherwise.
```
Comment (ihsandemir) [inline hazelcast-client/src/main/java/com/hazelcast/client/connection/ClientConnectionManager.java]: thread-safe

File: hazelcast-client/src/main/java/com/hazelcast/client/connection/ClientConnectionManager.java (hunk 1 above)
Comment (pveentjer) [reply hazelcast-client/src/main/java/com/hazelcast/client/connection/ClientConnectionManager.java]: This doesn't make a lot of sense. 

The javadoc in this PR makes it clear what is happening. Either the found connection is returned, or asynchronously the connection is made and null is returned.

File: hazelcast-client/src/main/java/com/hazelcast/client/connection/ClientConnectionManager.java (hunk 4 above)
Comment (pveentjer) [reply hazelcast-client/src/main/java/com/hazelcast/client/connection/ClientConnectionManager.java]: What does that mean? That I should add thread safe to each of these methods?

**Pull request closed without merging**